test/uk/gov/hmrc/agentservicesaccount/controllers/desiDetails/ContactDetailsControllerSpec.scala

```diff
@@ -379,21 +382,21 @@ class ContactDetailsControllerSpec extends UnitSpec with Matchers with GuiceOneA
       def shouldRedirect(endpoint: Action[AnyContent]): Unit = {
         val result = endpoint(fakeRequest())
         status(result) shouldBe SEE_OTHER
-        header("Location", result) shouldBe Some(routes.ContactDetailsController.showCurrentContactDetails.url)
+        header("Location", result) shouldBe Some(desiDetails.routes.ContactDetailsController.showCurrentContactDetails.url)
       }
 
       pendingChangesExistInRepo()
-      shouldRedirect(controller.showChangeBusinessName())
-      shouldRedirect(controller.submitChangeBusinessName())
-      shouldRedirect(controller.showChangeEmailAddress())
-      shouldRedirect(controller.submitChangeEmailAddress())
-      shouldRedirect(controller.finishEmailVerification())
-      shouldRedirect(controller.showChangeTelephoneNumber())
-      shouldRedirect(controller.submitChangeTelephoneNumber())
-      shouldRedirect(controller.startAddressLookup())
-      shouldRedirect(controller.finishAddressLookup(None))
-      shouldRedirect(controller.showCheckNewDetails)
-      shouldRedirect(controller.submitCheckNewDetails)
+      shouldRedirect(contactDetailsController.showChangeBusinessName())
```

Comment:
should we have a check that we cannot navigate to the new select changes when there are pending changes also?

Reply:
I don't think so as the idea is that the user should be able to navigate back and forwards through the journey, including back to the select changes page. The routing is designed so that if they resubmit with different options we will delete the session data for any unselected options - so shouldn't be a problem for them to go back to this page and resubmit. If I am misunderstanding let me know

Reply:
added